src/mero_conf_monitor.erl

```diff
@@ -39,28 +39,26 @@
 
 -type state() :: #state{}.
 -type cluster_config() :: mero:cluster_config().
--type init_args() :: #{orig_config | processed_config := cluster_config()}.
+-type init_args() :: #{orig_config := cluster_config()}.
 
 %%%-----------------------------------------------------------------------------
 %%% API
 %%%-----------------------------------------------------------------------------
--spec start_link(cluster_config(), cluster_config()) -> {ok, pid()} | {error, term()}.
-start_link(OrigConfig, ProcessedConfig) ->
-    gen_server:start_link({local, ?MODULE},
-                          ?MODULE,
-                          #{orig_config => OrigConfig, processed_config => ProcessedConfig},
-                          []).
+-spec start_link(cluster_config()) -> {ok, pid()} | {error, term()}.
+start_link(OrigConfig) ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, #{orig_config => OrigConfig}, []).
 
 %%%-----------------------------------------------------------------------------
 %%% Interesting Callbacks
 %%%-----------------------------------------------------------------------------
 -spec init(init_args()) -> {ok, state()}.
-init(#{orig_config := OrigConfig, processed_config := ProcessedConfig}) ->
+init(#{orig_config := OrigConfig}) ->
     program_heartbeat(),
+    self() ! heartbeat, % For the initial load
```

Comment:
this would let two `heartbeat`  msgs flying around each mero_conf:monitor_heartbeat_delay/1.    Perhaps remove the program_heartbeat/1 call there.

Reply:
Why not return `{ok, #state{…}, {continue, Something}}` and use `handle_continue/2` callback instead of sending a message to `self()`?

Reply:
TIL. Nice, I did that, good call.